models/0.py

```diff
@@ -32,6 +32,9 @@
 settings.academy_mode = True
 settings.lti_only_mode = False
 settings.coursera_mode = False
+# True to enable a testing environment, where feedback on certain question types
+# is not provided to the student.
+settings.is_testing = False
```

Comment:
settings is a site-wide variable that will effect all books currently being served.  Is this is a good strategy in a context where we are going to be serving many books simultaneously...  This seems like something that should be set and included as a part of the book rendering process, for a particular page...

Reply:
That's a good point. Actually, it opens up a whole set of questions:

1. How should feedback hiding work?
    1. Statically -- feedback is either always hidden or always shown until the student submits another answer or the page is refreshed.
    2. Dynamically -- the instructor can show/hide feedback, and students will see immediate updates. Doing this means that adding server push ability, a level of complexity I want to avoid.
2. What granularity should we support?
    1. Site-wide -- the current approach. I currently have a `runestone` application with all answers shown, and another `exam_runestone` application with all answers hidden. Both applications use the same course name, which means there's no extra steps for me to duplicate enrollment between a normal class and a class used only for exams.
    2. Book-wide -- That seems reasonable. However, it breaks my model above -- now, I need to create another course with the same enrollment as the non-exam course.
    3. Chapter/subchapter/page-wide -- sounds nice. If there was an exam "chapter", I could make all problems there hide feedback.
    4. Per-problem -- the most flexible.
3. Specifying hidden vs. revealed feedback
    1. As an additional parameter in the declaration of a problem. It would be a pain to go and change this for every problem type we support.
    2. Via some GUI where instructors could show/hide feedback in the admin interface. This appeals the most to me.
4. Security
    1. For low-stakes assessments -- in-class exercises where you don't want to give feedback until students have already worked the problem; hiding feedback before an assignment is due.
    2. High-stakes assessments -- provide some means to ensure student's can't even see the problems before they begin the exam. I do this my requiring an auth code only TAs know; TAs proctor the exam to help reduce cheating.

So, we need to discuss what makes sense. This PR implements a static, site-wide, high-stakes variant.

Reply:
I think there is a lot more to this that we want to consider before moving forward with this PR.

I'd like to see us spend the time actually designing this, either in an issue as I did for the reveal directive or on the wiki.

There are a lot of different use cases and a lot of different levels of security and a number of different deployment scenarios to figure out.

There is also an issue to consider around how much of the timed_exam directive we can leverage.  At a minimum it can benefit from some of this work, and we should not duplicate work that is already done there.

Reply:
We may want to consider whether some set of flags applies to feedback around all of the components when in dynamic mode with respect to server side versus client side checking of answers.

Reply:
I agree. Would you like to move this to an issue? Or just use this PR as a place to discuss? (The PR as is doesn't make sense to merge -- whatever it becomes, it will be very different.)

Reply:
Lets close this and move it to an issue.